src/factory/mastercopyDeployer.ts

```diff
@@ -1,57 +1,123 @@
+};
+
+/**
+ * Compute a module's mastercopy address. Where it is or will be deployed. And checks if it is already deployed.
+ *
+ * @param hre hardhat runtime environment
+ * @param mastercopyContractFactory mastercopy to get address for
+ * @param args the arguments passed to the mastercopy's constructor
+ * @returns {
+ *  address: string; // the address where the module mastercopy will be deployed or was already deployed
+ *  isDeployed: boolean; // true if the module mastercopy was already deployed on this chain
+ * }
+ */
+export const computeTargetAddress = async (
+  hre: HardhatRuntimeEnvironment,
+  mastercopyContractFactory: ContractFactory,
+  args: Array<any>,
+  salt: string
+): Promise<{ address: string; isDeployed: boolean }> => {
+  const deploymentTx = mastercopyContractFactory.getDeployTransaction(...args);
+  const singletonFactory = await getSingletonFactory(hre);
+
+  if (!Array.isArray(deploymentTx.data) || deploymentTx.data.length === 0) {
+    throw new Error("Unable to create the deployment data (no init code).");
+  }
+
+  const initCodeHash = keccak256(deploymentTx.data);
+
+  const computedAddress = getCreate2Address(
+    singletonFactory.address,
+    salt,
+    initCodeHash
+  );
+
+  const targetAddress = getAddress(
+    (await singletonFactory.callStatic.deploy(
+      deploymentTx.data,
+      salt
+    )) as string
+  );
+
+  // Sanity check
+  assert(
+    computedAddress === targetAddress || targetAddress === AddressZero,
+    "The computed address does not match the target address and the target address is not 0x0."
+  );
+
+  return {
+    address: computedAddress,
+    isDeployed: targetAddress === AddressZero,
+  };
 };
 
 export const deployMastercopyWithInitData = async (
   hre: HardhatRuntimeEnvironment,
   initCode: BytesLike,
   salt: string
-) => {
+): Promise<string> => {
   const singletonFactory = await getSingletonFactory(hre);
 
-  const targetAddress = await singletonFactory.callStatic.deploy(
-    initCode,
-    salt
+  // throws if this for some reason is not a valid address
+  const targetAddress = getAddress(
+    (await singletonFactory.callStatic.deploy(initCode, salt)) as string
   );
 
-  const initCodeHash = await hre.ethers.utils.solidityKeccak256(
-    ["bytes"],
-    [initCode]
-  );
-  const computedTargetAddress = await hre.ethers.utils.getCreate2Address(
+  const initCodeHash = keccak256(initCode);
+
+  const computedTargetAddress = getCreate2Address(
     singletonFactory.address,
     salt,
     initCodeHash
   );
 
-  if (targetAddress == "0x0000000000000000000000000000000000000000") {
+  if (targetAddress === AddressZero) {
     console.log(
       `        ✔ Mastercopy already deployed to: ${computedTargetAddress}`
     );
-    return;
+    return AddressZero;
   }
 
+  // Sanity check
+  assert.equal(
+    targetAddress,
+    computedTargetAddress,
+    "The computed address does not match the target address."
+  );
+
```

Comment:
I did not know this assert variation.

Usually I import from "assert"

And then simply do assert(a === b)